lib/Sema/TypeCheckType.cpp

```diff
@@ -3725,13 +3720,13 @@ Type swift::resolveCustomAttrType(CustomAttr *attr, DeclContext *dc,
 
   ASTContext &ctx = dc->getASTContext();
   if (TypeChecker::validateType(ctx, attr->getTypeLoc(), resolution, options))
-    return Type();
+    return ErrorType::get(ctx);
```

Comment:
The type returned from `validateType` should never be null. You can just propagate the type in the `TypeLoc` here. This'll help us kill off `TypeLoc` as a construct eventually.

Reply:
return attr->getTypeLoc().getType() instead in the check on validateType(). We still need the check though right? It looks like it might be safe to just call validateType and continue but I'm not 100% because I may be missing some mechanics.

Reply:
`validateType` is just returning whether the type in the typeloc has an error type in it anyways. You should still early-exit, just early-exit with `attr->getTypeLoc().getType()` instead of a new error type.

Reply:
I just pushed a patch that addresses this one.